### PR TITLE
Add LG telemetry

### PR DESCRIPTION
**Pull request by Danieladu**
ref： #2783
add LG telemetry client.

### CODE REVIEW

**Comment by Danieladu** [comment]
I found IBotTelemetryClient would be obsolete in this pr #3440.
Should this PR wait for the above PR and use the new class LogTelemetryClient?

**Comment by vishwacsena** [comment]
@Danieladu yes, this should follow what @garypretty  is doing for all up telemetry/ adaptive telemetry. @garypretty can you please add relevant PR #s to this for reference?

**Comment by garypretty** [comment]
We are just finalising the details of the other PR today. Once we have done that I'll reflect here.

**Comment by garypretty** [comment]
@Danieladu @vishwacsena IBotTelemetryClient will now remain. The decision has been taken to extend this interface.

**Comment by vishwacsena** [comment]
@garypretty can you take a look and approve if this is in line with your changes to instrument adaptive dialogs?

**Comment by vishwacsena** [comment]
This is the adaptive PR https://github.com/microsoft/botbuilder-dotnet/pull/3491/files

File: libraries/Microsoft.Bot.Builder.LanguageGeneration/Microsoft.Bot.Builder.LanguageGeneration.csproj [hunk 1]
```diff
@@ -46,6 +46,7 @@
 
   <ItemGroup>
     <ProjectReference Include="..\AdaptiveExpressions\AdaptiveExpressions.csproj" />
+    <ProjectReference Include="..\Microsoft.Bot.Builder\Microsoft.Bot.Builder.csproj" />
```
Comment (boydc2014) [inline libraries/Microsoft.Bot.Builder.LanguageGeneration/Microsoft.Bot.Builder.LanguageGeneration.csproj]: What we want do in R8 is to log in Adaptive only, not logging inside LG package.

**Comment by boydc2014** [comment]
We should just leverage the same telemetry client inside action (IDialog) and log the generation behavior in dialog level, and leave LG alone for now

File: libraries/Microsoft.Bot.Builder.Dialogs.Adaptive/Input/OAuthInput.cs [hunk 2]
```diff
@@ -238,6 +238,12 @@ public override async Task<DialogTurnResult> ContinueDialogAsync(DialogContext d
                         if (this.DefaultValueResponse != null)
                         {
                             var response = await this.DefaultValueResponse.BindToData(dc.Context, dcState).ConfigureAwait(false);
+                            var properties = new Dictionary<string, string>()
+                            {
+                                { "template", JsonConvert.SerializeObject(this.DefaultValueResponse) },
+                                { "result", response == null ? string.Empty : JsonConvert.SerializeObject(response, new JsonSerializerSettings() { NullValueHandling = NullValueHandling.Ignore }) },
+                            };
+                            TelemetryClient.TrackEvent("generatorResult", properties);
```
Comment (boydc2014) [inline libraries/Microsoft.Bot.Builder.Dialogs.Adaptive/Input/OAuthInput.cs]: probably user upper case "GeneratorResult"?

**Pull request merged**